test/onnx/verify/softmaxcrossentropyloss_no_attr.cpp

```diff
@@ -0,0 +1,185 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/register_target.hpp>
+#include <migraphx/verify.hpp>
+#include <onnx_test.hpp>
+
+TEST_CASE(softmaxcrossentropyloss_2d_no_reduction_test_ones)
+{
+    migraphx::program p = optimize_onnx("softmaxcrossentropyloss_2d_no_reduction_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape score_shape{migraphx::shape::float_type, {4, 4}};
+    std::vector<float> score_data(score_shape.elements(), 1);
+    migraphx::shape label_shape{migraphx::shape::int32_type, {4}};
+    std::vector<int32_t> label_data = {0, 3, 1, 2};
+
+    migraphx::parameter_map pp;
+    pp["0"] = migraphx::argument(score_shape, score_data.data());
+    pp["1"] = migraphx::argument(label_shape, label_data.data());
+
+    auto result = p.eval(pp).back();
+    std::vector<float> result_vector;
+    result.visit([&](auto output) { result_vector.assign(output.begin(), output.end()); });
+    std::vector<float> gold = {1.38629f, 1.38629f, 1.38629f, 1.38629f};
+
+    EXPECT(migraphx::verify::verify_rms_range(result_vector, gold));
+}
+
+TEST_CASE(softmaxcrossentropyloss_2d_no_reduction_test_asym_test)
+{
+    migraphx::program p = optimize_onnx("softmaxcrossentropyloss_2d_no_reduction_asym_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape score_shape{migraphx::shape::float_type, {3, 4}};
+    std::vector<float> score_data = {1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3};
+    migraphx::shape label_shape{migraphx::shape::int32_type, {3}};
+    std::vector<int32_t> label_data = {0, 3, 1};
+
+    migraphx::parameter_map pp;
+    pp["0"] = migraphx::argument(score_shape, score_data.data());
+    pp["1"] = migraphx::argument(label_shape, label_data.data());
+
+    auto result = p.eval(pp).back();
+    std::vector<float> result_vector;
+    result.visit([&](auto output) { result_vector.assign(output.begin(), output.end()); });
+    std::vector<float> gold = {1.38629436, 1.38629436, 1.38629436};
+
+    EXPECT(migraphx::verify::verify_rms_range(result_vector, gold));
+}
+
+TEST_CASE(softmaxcrossentropyloss_2d_no_reduction_test_zeros)
+{
+    migraphx::program p = optimize_onnx("softmaxcrossentropyloss_2d_no_reduction_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape score_shape{migraphx::shape::float_type, {4, 4}};
+    std::vector<float> score_data(score_shape.elements(), 0);
```

Comment:
This test has very limited value, IMO. It is same as testing everything with the value of `1`. Unless you could perhaps explain why it is uniquely valuable to test this value..! Thanks.

Reply:
This was created as I took a TDD approach to this operator. 

Valuable to test as for softmax giving us another sane known result as the input of zero is that it becomes an 1/N 
as softmax =  [ exp(0) / sum (exp(0)], and N is size of the input vector. With all 1's  you also get  each entry being (e/N(e)) which also simplifies to 1/N for each value. 

This ensures that these two mathematical cases are logically consistent and the operator has no issues with the softmax stage before any complex indexing (2 dimensions or above). I found it useful when using this and checking with MIGRAPHX_TRACE_EVAL=3 to ensure all outputs are updated consistently.

Although it appears trivial its another known point that can be checked by hand and ensure that all numbers will be populated correctly output and give the same correct value. If I used random data only, and got say a zero on output, it wouldn't be clear that a zero is due to the operator putting the right output, or an error with indexing if say that output wasn't populated.

The next test this goes with is with input values that are different, basically an std::iota(0, N) to ensure that each item is indexed to the correct position in the output vector. If that failed, I could use this test and the ones case to ensure with mathematically certainty that it isn't some error numerically that I was seeing.

Reply:
If you want to test special cases like `0` or `1`, then please  just set `score_data` elements to `[[0,0,0,0],[1,1,1,1][0.5,0.5,0.5,0.5],[x,x,x,x]]`, as an example. And you can verify all kind of combinations in just one test. Thanks.

Reply:
What I would recommend is a test beyond the all `1` or `0` case. Please test a test vector like `[[1,0,0,0], [0,1,0,0],[0,0,1,0],[0,0,0,1]]` whose entropy loss is calculated against labels like `[1,1,1,1]`. Thanks.

Reply:
> What I would recommend is a test beyond the all `1` or `0` case. Please test a test vector like `[[1,0,0,0], [0,1,0,0],[0,0,1,0],[0,0,0,1]]` whose entropy loss is calculated against labels like `[1,1,1,1]`. Thanks.

Please note, this is a one-hot test sample. It isn't the same as some new tests that you have added -- where first four indices are of one value, next 4 of another value, and last 8 values were filled with iota.

Reply:
Oh you wanted a one hot case. Okay. I added the previous of 0000, 1111, a,b,c,d ,e,f,g,h,i
Let me add that case as well

Reply:
Test is added and called 

softmaxcrossentropyloss_2d_no_reduction_one_hot_test

Reply:
Thanks.